### DISCUSSION TITLE
Contract Metadata Requirements

### DISCUSSION THREAD

**Discussion by oysterpack** [near/NEPs#155] [Ideas]
# Introduction
Contract metadata has been a hot topic of discussion. There is general consensus in the community that there is a strong need for contract metadata. However there is no consensus on the solution. The purpose of this discussion is to capture requirements.

Only after we can agree on requirements, then we can start working towards a solution (in a separate discussion).

### Side Notes
- [Fungible Token Metadata][1] is WIP but close to being finalized

[1]: https://github.com/near/NEPs/discussions/148#discussioncomment-366177

**Comment by robert-zaremba** [comment]
So the scope of this discussion is a general Contract Metadata? I don't think we can have some simple solution for that. The scope is too vast and we will end up updating it. That's why I prefer a _registry_ patter as outlined in #154 .

**Reply by oysterpack** [reply]
the purpose of this discussion is to collect requirements for contract metadata to understand the problem domain - registry pattern is a solution
